extensions/ql-vscode/src/mocks/request-handlers.ts

```diff
@@ -0,0 +1,137 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { DefaultBodyType, MockedRequest, rest, RestHandler } from 'msw';
+import { GetVariantAnalysisRepoRequest, GetVariantAnalysisRepoResultRequest, GitHubApiRequest, RequestKind } from './gh-api-request';
+
+const baseUrl = 'https://api.github.com';
+
+export type RequestHandler = RestHandler<MockedRequest<DefaultBodyType>>;
+
+export async function createRequestHandlers(scenarioDirPath: string): Promise<RequestHandler[]> {
+  const requests = await readRequestFiles(scenarioDirPath);
+
+  const handlers = [
+    createGetRepoRequestHandler(requests),
+    createSubmitVariantAnalysisRequestHandler(requests),
+    createGetVariantAnalysisRequestHandler(requests),
+    ...createGetVariantAnalysisRepoRequestHandlers(requests),
+    ...createGetVariantAnalysisRepoResultRequestHandlers(requests),
+  ];
+
+  return handlers;
+}
+
+async function readRequestFiles(scenarioDirPath: string): Promise<GitHubApiRequest[]> {
+  const files = await fs.readdir(scenarioDirPath);
+
+  const orderedFiles = files.sort((a, b) => {
+    const aNum = parseInt(a.split('-')[0]);
+    const bNum = parseInt(b.split('-')[0]);
+    return aNum - bNum;
+  });
+
+  const requests: GitHubApiRequest[] = [];
+  for (const file of orderedFiles) {
+    const filePath = path.join(scenarioDirPath, file);
+    const request: GitHubApiRequest = await fs.readJson(filePath, { encoding: 'utf8' });
+    requests.push(request);
+  }
+
+  return requests;
+}
+
+function createGetRepoRequestHandler(requests: GitHubApiRequest[]): RequestHandler {
+  const getRepoRequests = requests.filter(r => r.request.kind === RequestKind.GetRepo);
+
+  if (getRepoRequests.length > 1) {
+    throw Error('More than one get repo request found');
+  }
+
+  const getRepoRequest = getRepoRequests[0];
+
+  return rest.get(`${baseUrl}/repos/*`, (_req, res, ctx) => {
```

Comment:
I'm worried that this might result in matching too many URLs if we ever change the implementation. Can we make this a more specific match, either using the regex of the recorder or by using something like `${baseUrl}/repos/:owner/:name`?